src/views/News/index.js

```diff
@@ -43,465 +70,556 @@ const styles = {
+   */
+  function onDropAccepted(fileList) {
+    var previewImageFile = fileList[0];
+    var reader = new FileReader();
+    reader.onload = () => {
+      imageOnLoadHelper(reader);
+    };
+    reader.readAsDataURL(previewImageFile);
   }
 
-  updateSnackbar(message) {
-    this.setState({ snackbarMessage: message });
-    this.setState({ snackbarOpen: true });
+  /**
+   * Handles the rejection of the user dropping an invalid file in the Photo Updater Dialog Box
+   * Copied from Identification
+   */
+  async function onDropRejected() {
+    await clearPhotoDialogErrorTimeout();
+    setPhotoDialogError('Sorry, invalid image file! Only PNG and JPEG images are accepted.');
   }
 
   // Handles the actual 'edit' submission
-  async handleUpdate() {
-    let newsID = this.state.currentlyEditing;
+  async function handleUpdate() {
+    let newsID = currentlyEditing;
+
+    let imageData = null;
+
+    if (cropperRef.current !== null) {
+      imageData = cropperRef.current.cropper
+        .getCroppedCanvas({ width: CROP_DIM })
+        .toDataURL()
+        .replace(/data:image\/[A-Za-z]{3,4};base64,/, '');
+    } else if (cropperImageData !== null) {
+      imageData = cropperImageData.replace(/data:image\/[A-Za-z]{3,4};base64,/, '');
+    }
+
     // create the JSON newData object to update with
     let newData = {
-      categoryID: this.state.newPostCategory,
-      Subject: this.state.newPostSubject,
-      Body: this.state.newPostBody,
+      categoryID: newPostCategory,
+      Subject: newPostSubject,
+      Body: newPostBody,
+      Image: imageData,
     };
 
     // update the news item and give feedback
     let result = await newsService.editStudentNews(newsID, newData);
     if (result === undefined) {
-      this.updateSnackbar('News Posting Failed to Update');
+      createSnackbar('News Posting Failed to Update', 'error');
     } else {
-      this.updateSnackbar('News Posting Updated Successfully');
+      createSnackbar('News Posting Updated Successfully', 'success');
     }
 
-    // close the window and reload to update data
-    // (necessary since data is currently not pulled from render method)
-    this.setState({ openPostActivity: false });
-    window.top.location.reload();
+    setOpenPostActivity(false);
+    loadNews(); //reload news
   }
 
-  async handleSubmit() {
-    // create the JSON newsItem object to post
+  async function handleSubmit() {
+    let newImage;
+
+    if (cropperImageData !== null) {
+      let croppedImage = cropperRef.current.cropper
+        .getCroppedCanvas({ width: CROP_DIM })
+        .toDataURL();
+      newImage = croppedImage.replace(/data:image\/[A-Za-z]{3,4};base64,/, '');
+    }
+
+    //There is no else here because if there is no image, the JSON object SHOULD
+    //contain null, since the API has been designed to look for a null case and handle
+    //it, treating it appropriately to mean there is no image for the post.
```

Comment:
I'm not sure this comment is necessary. I find it more confusing than helpful when making sense of this code.